src/util/isoDateTime/isoDateTime.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { isISODateTime } from './isoDateTime'
+
+describe('isISODateTime', () => {
+  test.each([
+    ['2023', false],
+    ['2023-01', false],
+    ['2023-01-01', false],
+    ['2023-01-01T00:00', false],
+    ['2023-01-01T00:00:00', true],
+    ['2023-01-01T00:00:00.000', true],
+    ['2023-01-01T00:00:00.000Z', true],
+    ['2023-01-01T00:00:00.000+01:00', true],
+    ['2023-01-01T00:00:00.000-01:00', true],
+    ['002023-01-01T00:00:00.000+01:00', false]
+  ])('validates input %s as %s', (input: string, expected: boolean) => {
+    expect(isISODateTime(input)).toBe(expected)
```

Comment:
IMHO you should only test the `convertToYearIfISODateTime` method that will test all other methods in a traversal way and ensures you 100% of coverage.

Reply:
Thanks for pointing this out. That makes sense to get a full coverage.